### PR TITLE
Add WebSocket command bridge and Laravel panel API

### PR DESCRIPTION
**Pull request by atomicdeploy**
## Summary
- add WP-CLI WebSocket server exposed through `/wordpress-ws`
- share Digitalogic commands across AJAX, WebSocket, REST, and Laravel panel bridge
- add admin-wide `admin-ajax.php` WebSocket proxy with fallback and opt-out
- support registered `wp_ajax_{action}` callbacks, including slash-containing plugin actions
- migrate SMS.ir and comment guard mu-plugin code into plugin classes

## Verification
- `node --check assets/js/ajax-proxy.js`
- `node --check assets/js/admin.js`
- `wp digitalogic products list --limit=1 --format=json --allow-root`
- WebSocket smoke tests for `digitalogic_get_products` and a generic WooCommerce AJAX action
- Laravel bridge HTTPS check returned 200 with product data
- `git diff --check`

## Notes
- `php -l` currently segfaults on this server even for a one-line PHP file, so PHP validation used WP-CLI runtime loading.
- The WebSocket proxy skips FormData/uploads, selected core actions such as `heartbeat`, and any jQuery request with `digitalogicWebSocket: false`.

### CODE REVIEW

**Comment by atomicdeploy** [comment]
Production verification update:

- Fixed the live /wordpress-ws route on Apache by proxying it to the local WP-CLI WebSocket service on 127.0.0.1:8090.
- Added and enabled digitalogic-wordpress-ws.service; systemctl is-active digitalogic-wordpress-ws.service returns ctive.
- Installed websocat 1.14.1 at /usr/local/bin/websocat and verified the public socket path is no longer a WordPress 404.
- Blank/unauthenticated probe: wss://digitalogic.ir/wordpress-ws?nonce= now returns HTTP/1.1 403 Forbidden, which is expected auth rejection rather than not found.
- WebSocket command parity test using digitalogic_get_products and limit=1 returned: { success: true, count: 1, total: 222, first_id: 11303}.
- Matching wp-admin/admin-ajax.php POST for the same action returned the same compact result: {success: true, count: 1, total: 222, first_id: 11303}.
- Laravel panel launcher/handoff was tested from a real WordPress admin session: redirect host panel.digitalogic.ir, one-time code length 48, /wp-json/digitalogic-panel/v1/session/consume returned user 1, manage_woocommerce: true, theme Digitalogic, and shared UI base https://digitalogic.ir/digitalogic-ui/.
- PHP syntax checks pass for changed PHP files with opcache.enable_cli=0; the server's default PHP 8.5 CLI currently segfaults on php -l with OPcache/ionCube loaded, so I avoided treating that extension-stack crash as a source syntax failure.
- 
ode --check assets/js/ajax-proxy.js and 
ode --check assets/js/admin.js pass.
- pachectl configtest returns Syntax OK.

Caveat: I did not find a Laravel app (rtisan) under /var/www, and panel.digitalogic.ir does not appear to be backed by a running local Laravel service yet. This PR wires the WordPress plugin side: admin entry point, one-time WordPress auth handoff, REST consume endpoint, command bridge, and shared theme metadata for Laravel to consume once the panel app is deployed.

**Comment by atomicdeploy** [comment]
Follow-up production/browser verification update:

- Stopped using the reserved panel.digitalogic.ir default. The temporary working panel now lives at https://digitalogic.ir/panell/ and is served by the plugin.
- Added a Vue.js SPA panel with Persian/English labels, RTL/LTR behavior, system/light/dark modes, and history routing such as /panell/products/11307.
- The panel renders without the frontend theme wrapper and uses the current WordPress login.
- Browser verified with the in-app browser:
  - /panell/ redirects unauthenticated users to wp-login.php.
  - Authenticated /panell/products loads products.
  - The panel shows WebSocket connected in Persian (وب سوکت متصل است).
  - Clicking a product pushes /panell/products/11307 without a full reload.
  - Browser back returns to /panell/products/.
  - Direct deep link /panell/products/11307 renders the product detail page.
  - Console warnings/errors were empty during these checks.
- Fixed the admin-wide WebSocket AJAX proxy response shape so proxied admin-ajax calls emulate the normal {success, data} WordPress response envelope.
- Verified the old WordPress Products page no longer shows the invalid response alert; it renders 50 rows.
- Replaced the Products table View button with a real translated link to /panell/products/{id} in a new tab, avoiding the forbidden core post.php?post=...&action=edit product editor route.
- Added short-lived per-user browser WebSocket tokens, while keeping the server token for service/websocat checks. This avoids depending on cookie forwarding across the WebSocket proxy path.
- Added optional direct local Laravel loading bridge endpoints. They remain inert unless digitalogic_laravel_app_path is configured, so no reserved vhost or platform rewrite is touched.

Final checks:

- 
ode --check passed for ssets/js/panel-app.js, ssets/js/ajax-proxy.js, and ssets/js/admin.js.
- PHP lint passed with opcache.enable_cli=0 for changed PHP files.
- digitalogic-wordpress-ws.service is active.
- websocat command parity still returns { success: true, count: 1, total: 222, first_id: 11303}.
- Matching wp-admin/admin-ajax.php parity still returns {success: true, count: 1, total: 222, first_id: 11303}.
- Laravel local status endpoint returns {success:true,configured:false,available:false} until an app path is configured.

**Comment by atomicdeploy** [comment]
Mergeability update after syncing with main:

- Merged origin/main into this branch and resolved conflicts in ssets/js/admin.js and includes/admin/class-admin.php.
- Kept main's server-side DataTables pagination while preserving the WebSocket/admin-ajax response normalization and /panell/products/{id} View links.
- Restored 	otal as an alias beside 
ecordsTotal/
ecordsFiltered so WebSocket and admin-ajax parity remains intact.
- Re-ran browser admin Products verification after the merge: 50 rows, 50 panel View links, second link points to https://digitalogic.ir/panell/products/11307, no console errors/warnings.
- Re-ran protocol checks: WebSocket and admin-ajax both return { success: true, count: 1, total: 222, first_id: 11303}.
- GitHub now reports mergeStateStatus: CLEAN.
- CI completed: Code Quality Check, PHP 8.3 Test, and Security Check all succeeded. Build Plugin was skipped by workflow rules.

**Comment by atomicdeploy** [comment]
Panel polish update pushed in ef0e0dd.

Verified on the live site:
- /panel/ is now canonical; /panell/... redirects to /panel/...
- WebSocket service is active, and websocat returns the same product command shape as wp-admin admin-ajax: success=true, total=222, first_id=11303
- Browser-tested the live /panel/ UI: authenticated admin user is shown, WebSocket status is connected, USD/CNY are number-formatted, dashboard widgets have action/reorder controls, product search filters instantly from the local cache, and SPA navigation updates URLs for /panel/cli and /panel/sync
- WP-CLI command guidance and atomicdeploy/patris-export sync guidance are visible in the panel

Local/server checks:
- node --check assets/js/panel-app.js assets/js/admin.js
- php -l on touched PHP files
- git diff --check
- GitHub CI is passing: Code Quality Check, PHP 8.3 Test, Security Check

**Comment by atomicdeploy** [comment]
UX/report migration update pushed in 41ed9ae.

Live Browser verification:
- Dashboard now shows short connection text (متصل) with a dot indicator, not the old WebSocket icon/text.
- /panel dashboard has inline USD/CNY editing and save controls instead of redirecting to settings.
- The old placeholder “still missing” text was removed from the UI and code; dashboard/reports now show six migrated report-flow cards based on reportFinal: product price reports, price synchronization, image audit, customer reports, currency/shipping, and Excel exports.
- Modern/Classic style comparison selector works; Classic switches to compact 6px admin-style controls.
- Product search remains instant; AT25040B filtered to 1 / 222.
- Product table is display-first and Excel-like: no always-visible row inputs; clicking a cell opens a cell editor. Row padding is now 7px 10px.
- Reports navigation works at /panel/reports.

Code checks:
- node --check assets/js/panel-app.js assets/js/admin.js
- php -l includes/panel/class-panel.php includes/panel/views/app.php
- git diff --check
- GitHub CI passing: Code Quality Check, PHP 8.3 Test, Security Check

**Comment by atomicdeploy** [comment]
Follow-up architecture audit (2026-07-20): the Vue `/panel/` is a WordPress-first route, but no Laravel application (`artisan` or `bootstrap/app.php`) is present in this repository or on the production host. The token/session-handoff bridge added here therefore does not represent the now-confirmed target architecture.

- PR #87 fixes the live panel rewrite regression and stops same-origin launches from emitting a handoff code; WordPress session/capabilities are used directly.
- Issue #83 tracks bundling Laravel into this same repository/runtime, reusing WordPress auth/authZ and the existing command/WebSocket infrastructure, followed by removal of the legacy external token endpoints.

This note preserves the original implementation history while recording the required compatibility direction; no owner approval is implied.
